src/api/web3.js

```diff
@@ -0,0 +1,85 @@
+import { Deployer } from '@noblocknoparty/contracts'
+import Web3 from 'web3'
+
+import { DEPLOYER_CONTRACT_ADDRESS, NETWORK } from '../config'
+
+let web3
+export let networkError
+
+function getWeb3() {
+  if (!web3) {
+    return setupWeb3()
+  } else {
+    return web3
+  }
+}
+
+export async function getDeployerAddress() {
+  // if local env doesn't specify address then assume we're on a public net
+  const id = await web3.eth.net.getId()
+  return DEPLOYER_CONTRACT_ADDRESS || Deployer.NETWORKS[id].address
+}
+
+export async function getTransactionLogs(txHash) {
+  const web3 = getWeb3()
+
+  const logs = await web3.eth.getTransactionReceipt(txHash)
+
+  return logs
+}
+
+export async function getEvents(address, abi) {
+  return new Promise(function(resolve, reject) {
+    const Contract = window.web3.eth.contract(abi)
+    const instance = Contract.at(address)
+    const events = instance.allEvents({ fromBlock: 0, toBlock: 'latest' })
+    events.get(function(error, result) {
+      if (error) {
+        reject(error)
+      }
+
+      resolve(result)
+    })
+  })
+}
+
+export async function getAccount() {
+  const web3 = getWeb3()
+  const accounts = await web3.eth.getAccounts()
+  return accounts[0]
+}
+
+export async function setupWeb3() {
+  //Localnode
```

Comment:
I don't think we should try to connect directly to `localhost`. That's what a dapp browser / metamask is for. Just switch networks within Metamask - this way we also ensure we're testing our Metamask link up locally.

I think the flow we want is:

1. Check if web3.currentProvider is available. If so then use that.
2. If not then use the Infura read-only endpoint.

Reply:
Also, the config JSON will specify which network is expected. The setup method to check that we're connected to the expected network.